lib/rucio/client/uploadclient.py

```diff
@@ -621,23 +622,24 @@ def _upload_item(self, rse_settings, rse_attributes, lfn, source_dir=None, domai
         # Auth. mostly for object stores
         if sign_service:
             pfn = self.client.get_signed_url(rse_settings['rse'], sign_service, 'write', pfn)       # NOQA pylint: disable=undefined-variable
+            protocol_read = self._create_protocol(rse_settings, 'read', domain=domain, impl=impl)
+            readpfn = self.client.get_signed_url(rse_settings['rse'], sign_service, 'read', pfn)    # NOQA pylint: disable=undefined-variable
```

Comment:
I think it would be better to rename this to `signed_read_pfn` and the tmp below to `signed_read_pfn_tmp`. Just to make it absolutely clear in the code, that this is not a normal readpfn which might be needed otherwise, but it is really just for the signed urls.